test/library_search/basic.lean

```diff
@@ -70,9 +70,50 @@ by library_search -- says: `exact add_pos ha hb`
 example (a b : ℕ) : 0 < a → 0 < b → 0 < a + b :=
 by library_search -- says: `exact add_pos`
 
+section synonym
+
+-- Synonym `>` for `<` in the goal
+example (a b : ℕ) : 0 < a → 0 < b → a + b > 0 :=
+by library_search -- says: `exact add_pos`
+
+-- Synonym `>` for `<` in another part of the goal
+example (a b : ℕ) : a > 0 → 0 < b → 0 < a + b :=
+by library_search -- says: `exact add_pos`
+
+-- Synonym `>` for `<` in another part of the goal
+example (a b : ℕ) (ha : a > 0) (hb : 0 < b) : 0 < a + b :=
+by library_search -- says: `exact add_pos ha hb`
+
 example (a b : ℕ) (h : a ∣ b) (w : b > 0) : a ≤ b :=
 by library_search -- says: `exact nat.le_of_dvd w h`
 
+example (a b : ℕ) (h : a ∣ b) (w : b > 0) : b ≥ a :=
+by library_search -- says: `exact nat.le_of_dvd w h`
+
+-- A lemma with head symbol `¬` can be used to prove `¬ p` or `⊥`
+example (a : ℕ) : ¬ (a < 0) := by library_search -- says `exact not_lt_bot`
+
+example (a : ℕ) (h : a < 0) : false := by library_search -- says `exact not_lt_bot h`
+
+def P : ℕ → Prop := λ _, sorry
+
+-- This lemma with `>` as its head symbol should also be found for goals with head symbol `<`.
+lemma lemma_with_gt_in_head (a : ℕ) : P a → 0 > a := sorry
+
+example (a : ℕ) (h : P a) : 0 > a := by library_search -- says `exact lemma_with_gt_in_head a h`
+
+example (a : ℕ) (h : P a) : a < 0 := by library_search -- says `exact lemma_with_gt_in_head a h`
+
+lemma lemma_with_false_in_head (a b : ℕ) (h1 : a < b) (h2 : P a) : false := sorry
+
+set_option trace.silence_library_search false
+example (a b : ℕ) (h1 : a < b) (h2 : P a) : false := by library_search
+-- says `exact lemma_with_false_in_head a b h1 h2`
+
+example (a b : ℕ) (h1 : a < b) : ¬ (P a) := by library_search
+-- says `exact lemma_with_false_in_head a b h1`
+
```

Comment:
One test case that failed before the PR and still does, but might be not too much work to incorporate:
```lean
example (a b : ℕ) (h2 : P a) : ¬ (a < b) := by library_search -- currently fails
```